update_image.pl

```diff
@@ -112,10 +144,206 @@
+        # If no input bin file then the pnor script handles creating the content
+        if(!exists $sections{$section}{in})
+        {
+             # Build up the systemBinFiles argument
+             $system_bin_files .= "$separator$section=".EMPTY;
+        }
+        else
+        {
+            # Stage the input file
+            run_command("cp $sections{$section}{in} "
+             . "$scratch_dir/$section.staged");
+
+            # If secureboot compile, there can be extra protected
+            # and unprotected versions of the input to stage
+            if(-e "$sections{$section}{in}.protected")
+            {
+                run_command("cp $sections{$section}{in}.protected "
+                    . "$scratch_dir/$section.staged.protected");
+            }
+
+            if(-e "$sections{$section}{in}.unprotected")
+            {
+                run_command("cp $sections{$section}{in}.unprotected "
+                    . "$scratch_dir/$section.staged.unprotected");
+            }
+            # Build up the systemBinFiles argument
+            $system_bin_files .= "$separator$section=$scratch_dir/"
+                . "$section.staged";
+        }
+    }
+
+    if(length($system_bin_files))
+    {
+        # Point to the location of the signing tools
+        $ENV{'DEV_KEY_DIR'}="$ENV{'HOST_DIR'}/etc/keys/";
+        $ENV{'SIGNING_DIR'} = "$ENV{'HOST_DIR'}/usr/bin/";
+        $ENV{'SIGNING_TOOL_EDITION'} = "community";
+
+        # Determine whether to securely sign the images
+        my $securebootArg = $secureboot ? "--secureboot" : "";
+
+        # Determine whether a key transition should take place
+        my $keyTransitionArg = $key_transition ne "" ? "--key-transition $key_transition" : "";
+        # Determine which type of signing to use
+        my $signModeArg = $sign_mode ne "" ? "--sign-mode $sign_mode" : "";
+
+        # Process each image
+        my $cmd =   "cd $scratch_dir && "
+                  . "$hb_image_dir/genPnorImages.pl "
+                      . "--binDir $scratch_dir "
+                      . "--systemBinFiles $system_bin_files "
+                      . "--pnorLayout $pnor_layout "
+                      . "$securebootArg $keyTransitionArg $signModeArg "
+                      . "--hwKeyHashFile $hb_image_dir/imprintHwKeyHash";
+
+        # Print context not visible in the actual command
+        if($debug)
+        {
+            print STDOUT "SIGNING_DIR: " . $ENV{'SIGNING_DIR'} . "\n";
+            print STDOUT "DEV_KEY_DIR: " . $ENV{'DEV_KEY_DIR'} . "\n";
+            print STDOUT "SIGNING_TOOL_EDITION: "
+                . $ENV{'SIGNING_TOOL_EDITION'} . "\n";
+        }
+
+        run_command($cmd);
+
+        # Copy each output file to its final destination
+        foreach my $section (keys %sections)
+        {
+            next if(!exists $sections{$section}{in});
+            run_command("cp $scratch_dir/$section.bin "
+                . "$sections{$section}{out}");
+        }
+    }
```

Comment:
It's not hurting anything per se, other than total disk space; one nicety is it puts all the original files in the scratch even if they would otherwise never have been there, so it's easy to get at the original contents of everything without jumping into a bunch of other random directories.  They also don't accumulate as # of rebuilds increases.    My thought is leave for now, and if the space becomes a concern we can come back and delete everything pretty easily.